spec.md

```diff
@@ -0,0 +1,125 @@
+# Yearn Vesting Escrow
+
+version 0.3-dev0
+
+## Vesting Factory
+
+- the factory is initialized with a template contract of `Vesting Escrow`
+- `deploy_vesting_escrow`
+    - arguments
+        - `token: address` ERC20 funding token
+        - `recipient: address` vesting recipient
+            - can be a smart contract in v0.3
+        - `amount: uint256` total vesting amount
+        - `vesting_duration: uint256` in seconds
+        - `vesting_start: uint256 = block.timestamp` in unix epoch
+        - `cliff_duration: uint256 = 0` in seconds
+        - `open_claim: bool = True`
+            - allows anyone to claim to recipient, use this for smart contract recipients
+        - `owner: address = msg.sender`
+            - this party can terminate and clawback the escrow before it fully vests
+            - set to `empty(address)` to disable clawback, this can also be done later with `escrow`
+        - `support_vyper: uint256 = 100` in bps
+    - constraints
+        - `token`, `amount`
+            - `token` that doesn't return True on transfer is supported via `default_return_value=True`
+            - implicitly checked by `token` transfer
+                - the funder must have a `token` balance of at least `amount`
+                - the factory must have a `token` allowance of at least `amount`
+        - `vesting_start + vesting_duration`
+            - end time must be in the future
+            - a regular transfer would suit better if it's in the past
+        - `recipient`
+            - must not be zero address
+        - `owner`
+            - can be zero address
+        - `cliff_duration`
+            - must not exceed vesting duration
+        - `support_vyper`
+            - check we are on mainnet in case someone deploys the contract on other networks
+            - or ask vyper team to make a new multisig using create2 so they can resurrect it on any network
+    - actions
+        - create a new vesting escrow using `create_minimal_proxy_to`
+        - fund it by transferring tokens from `msg.sender` to the newly created `escrow`
+        - `initialize` the `escrow`
+        - log the creation parameters in `VestingEscrowCreated` event
+        - if `support_vyper > 0`, transfer additional tokens from `msg.sender` to `vyperlang.eth`
+
+
+## Vesting Escrow
+
+- global considerations
+    - return an appropriate value or True from methods to make contract calls cheaper
+    - use `default_return_value=True` for token transfers
+
+- `__init__`
+    - actions
+        - template contract must be made defunct by storing `initialized` as True
```

Comment:
Just in case, the architecture doesn't use blueprints, that right?

Reply:
yes, after looking at usage data, we decided against blueprint and reverted back to minimal proxies.